netcfgbu/config_model.py

```diff
@@ -35,121 +39,210 @@
+        """
         return Path(value).absolute()
 
-    @validator("plugins_dir")
-    def _plugins_dir(cls, value):  # noqa
+    @field_validator("plugins_dir", mode="after")
+    @classmethod
+    def _plugins_dir(cls, value: str) -> Path:  # noqa
+        """Convert the plugins directory path to an absolute path.
+
+        If the value is the current working directory and doesn't contain "/plugins",
+        append "/plugins" to the path.
+
+        Args:
+            value: The plugins directory path.
+
+        Returns:
+            Path: The absolute path of the plugins directory.
+        """
         if value == os.getenv("PWD") and "/plugins" not in value:
-            value = value + "/plugins"
+            value += "/plugins"
         return Path(value).absolute()
 
 
-class FilePathEnvExpand(FilePath):
-    """ A FilePath field whose value can interpolated from env vars """
-
-    @classmethod
-    def __get_validators__(cls):
-        yield from EnvExpand.__get_validators__()
-        yield from FilePath.__get_validators__()
+# A FilePath field whose value can be interpolated from env vars
+FilePathEnvExpand = Annotated[FilePath, BeforeValidator(expand_env_str)]
 
 
 class GitSpec(NoExtraBaseModel):
-    name: Optional[str]
+    """Represents the configuration for a Git repository.
+
+    Attributes:
+        name: Optional name for the Git configuration.
+        repo: The Git repository URL.
+        add_tag: Whether to add a tag or not.
+        email: Optional email for Git operations.
+        username: Optional username for Git authentication.
+        password: Optional password for Git authentication.
+        token: Optional token for Git authentication.
+        deploy_key: Optional deploy key file path for Git authentication.
+        deploy_passphrase: Optional passphrase for the deploy key.
+    """
+
+    name: Optional[str] = None
     repo: EnvExpand
-    email: Optional[str]
-    username: Optional[EnvExpand]
-    password: Optional[EnvExpand]
-    token: Optional[EnvSecretStr]
-    deploy_key: Optional[FilePathEnvExpand]
-    deploy_passphrase: Optional[EnvSecretStr]
-
-    @validator("repo")
-    def validate_repo(cls, repo):  # noqa
+    add_tag: Optional[bool] = False
+    email: Optional[str] = None
+    username: Optional[EnvExpand] = None
+    password: Optional[EnvExpand] = None
+    token: Optional[EnvSecretStr] = None
+    deploy_key: Optional[FilePathEnvExpand] = None
+    deploy_passphrase: Optional[EnvSecretStr] = None
+
+    @field_validator("repo")
+    @classmethod
+    def validate_repo(cls, repo: str) -> str:
+        """Validate the Git repository URL.
+
+        Args:
+            repo: The repository URL.
+
+        Returns:
+            str: The validated repository URL.
+
+        Raises:
+            ValueError: If the repository URL does not start with the expected protocols.
+        """
         expected = ("https:", "git@")
         if not repo.startswith(expected):
```

Comment:
**Suggestion:** The `validate_repo` method in the `GitSpec` class should handle the case where the repository URL is `None` or an empty string before checking its start. This prevents raising an unrelated error when the URL is not provided. [enhancement, importance: 8]
```suggestion
if not repo or not repo.startswith(expected):
```